### DISCUSSION TITLE
how to update dependent package

### DISCUSSION THREAD

**Discussion by Fak3** [pdm-project/pdm#3299] [Q&A]
My project depends on `httpx`:
```toml
dependencies = [ "httpx>=0.21",]
```
`httpx` depends on `httpcore`. Curretly installed version of the `httpcore`, same as pinned version in the lockfile: 0.13.2.
This `httpcore` is too old, and incompatible with `httpx` 0.25.1.

How can i update `httpcore` in my project?

I tried the following and it failed:
```
> pdm update --update-eager httpcore==1
[ProjectError]: httpcore==1 does not exist in default dependencies nor is a transitive dependency.

> pdm update --update-all httpcore==1
[ProjectError]: httpcore==1 does not exist in default dependencies nor is a transitive dependency.

> pdm update  --update-all httpx
Updating packages: httpx.
  0:00:19 🔒 Lock successful.  
All packages are synced to date, nothing to do.
  ✔ Update paradox 0.1 -> 0.1 successful

  0:00:00 🎉 All complete! 0/0
  
 > pdm update  --update-all httpcore
Updating packages: httpcore.
  0:00:30 🔒 Lock successful.  
All packages are synced to date, nothing to do.
  ✔ Update paradox 0.1 -> 0.1 successful

  0:00:00 🎉 All complete! 0/0
  ```

**Comment by Fak3** [comment]
Related #2628

**Comment by frostming** [comment]
You don't even try it:

```
pdm update httpcore
```

**Reply by Fak3** [reply]
I'm pretty sure i've tried it with the same result as `pdm update  --update-all httpcore`. I.e. it does nothing.
Curretly trying to reproduce, I ran into another bug #3319

**Reply by frostming** [reply]
Note you have pinned `h11` and `h2` in pyproject.toml, you can't get `httpcore` upgraded unless you loosen that constraints

**Reply by Fak3** [reply]
> Note you have pinned `h11` and `h2` in pyproject.toml, you can't get `httpcore` upgraded unless you loosen that constraints

I wonder if pdm could help to find unsatisfied constraints. For example if you run `pdm update httpcore==1` it would be insightful to know that pinned h11 is incompatible. Currently I can only learn that if I first explicitly pin httpcore in pyproject.toml

**Reply by Fak3** [reply]
I do realize that the issue I am having is that hhtpx did not properly set required dependency versions. So I can end up with httpx incompatible with httpcore.
